client/packages/system/src/utils.ts

```diff
@@ -158,3 +158,44 @@ export const getNameValue = (t: TypedTFunction<LocaleKey>, name: string) => {
   if (name == 'Inventory adjustments') return t('inventory-adjustment');
   return name;
 };
+
+export enum ReportType {
+  INBOUND = "Inbound Shipment",
+  INTERNAL_ORDER = "Internal Order",
+  OUTBOUND_INVOICE_LANDSCAPE_WITH_LOGO = "Outbound Shipment (Landscape)",
+  OUTBOUND_INVOICE_PORTRAIT_WITH_LOGO = "Outbound Shipment (Portrait)",
+  PRESCRIPTION_RECEIPT = "Prescription Receipt",
+  REPACK = "Repack",
+  REQUISITION = "Requisition",
+  STOCK_TAKE_DETAIL_VIEW = "Stocktake",
+  STOCKTAKE_VARIANCE = "Stocktake Variance",
+  STOCK_TAKE_WITH_QUANTITY = "Stocktake with Quantity",
+  STOCK_TAKE_WOTHOUT_QUANTITY = "Stocktake without Quantity",
+}
+
+export const REPORT_TRANSLATION_KEYS: Record<ReportType, LocaleKey> = {
+  [ReportType.STOCKTAKE_VARIANCE]: 'form.stocktake-variance',
+  [ReportType.INBOUND]: 'form.inbound',
+  [ReportType.INTERNAL_ORDER]: 'form.internal-order',
+  [ReportType.OUTBOUND_INVOICE_LANDSCAPE_WITH_LOGO]: 'form.outbound-invoice-landscape-woth-logo',
+  [ReportType.OUTBOUND_INVOICE_PORTRAIT_WITH_LOGO]: 'form.outbound-invoice-portrait-woth-logo',
+  [ReportType.PRESCRIPTION_RECEIPT]: 'form.prescription-receipt',
+  [ReportType.REPACK]: 'form.repack',
+  [ReportType.REQUISITION]: 'form.requisition',
+  [ReportType.STOCK_TAKE_DETAIL_VIEW]: 'form.stocktake-detail-view',
+  [ReportType.STOCK_TAKE_WITH_QUANTITY]: 'form.stocktake-with-quantity',
+  [ReportType.STOCK_TAKE_WOTHOUT_QUANTITY]: 'form.stocktake-without-quantity',
+};
+
+export const getReportKey = (reportName: string): LocaleKey => {
+  const reportType = Object.values(ReportType).find(
+    type => type === reportName
+  );
+  if (reportType) {
+    return REPORT_TRANSLATION_KEYS[reportType];
+  }
+  console.warn('Report name not found in translation keys:', reportName);
+  // Fallback to using the report name directly as a key
+  return reportName as LocaleKey;
+}
```

Comment:
All of this could be replaced by an 'as LocaleKey' in the translation function in report selector.

It adds work also to developers when making changes to report names for example - but also ensures that we have type safety for localekeys, and that report missing translations are easily visible.

Still safely falls back to report name when no key is found. This is also ignored when custom reports are used (as we would always fall back in this case as the name is already translated)